packages/smart-wallet/test/test-addAsset.js

```diff
@@ -87,3 +99,441 @@ test('avoid O(wallets) storage writes for a new asset', async t => {
+  assert(res, 'resolution is unexpectedly falsy');
+  return res;
+};
+
+/** @typedef {import('@agoric/internal/src/storage-test-utils.js').MockChainStorageRoot} MockVStorageRoot */
+
+const UNIT6 = 1_000_000n; // IST has six decimals
+const CENT = UNIT6 / 100n;
+
+/** @param {import('ava').ExecutionContext<Awaited<ReturnType<makeDefaultTestContext>>>} t */
+const makeScenario = t => {
+  /**
+   * A player and their user agent (wallet UI, signer)
+   *
+   * @param {string} addr
+   * @param {PromiseKit<*>} bridgeKit - to announce UI is ready
+   * @param {MockVStorageRoot} vsRPC - access to vstorage via RPC
+   * @param {typeof t.context.sendToBridge} broadcastMsg
+   * @param {*} walletUpdates - access to wallet updates via RPC
+   */
+  const aPlayer = async (
+    addr,
+    bridgeKit,
+    vsRPC,
+    broadcastMsg,
+    walletUpdates,
+  ) => {
+    const ctx = makeImportContext();
+    const vsGet = path =>
+      E(vsRPC).getBody(`mockChainStorageRoot.${path}`, ctx.fromBoard);
+
+    // Ingest well-known brands, instances.
+    // Wait for vstorage publication to settle first.
+    await eventLoopIteration();
+    /** @type {Record<string, Brand>} */
+    // @ts-expect-error unsafe testing cast
+    const wkBrand = Object.fromEntries(await vsGet(`agoricNames.brand`));
+    await vsGet(`agoricNames.instance`);
+    t.log(addr, 'wallet UI: ingest well-known brands', wkBrand.Place);
+
+    assert(broadcastMsg);
+    /** @param {import('../src/smartWallet.js').BridgeAction} action */
+    const signAndBroadcast = async action => {
+      const actionEncoding = ctx.fromBoard.toCapData(action);
+      const addIssuerMsg = {
+        type: ActionType.WALLET_SPEND_ACTION,
+        owner: addr,
+        spendAction: JSON.stringify(actionEncoding),
+        blockTime: 0,
+        blockHeight: 0,
+      };
+      t.log(addr, 'walletUI: broadcast signed message', action.method);
+      await broadcastMsg(addIssuerMsg);
+    };
+
+    const auxInfo = async obj => {
+      const slot = ctx.fromBoard.toCapData(obj).slots[0];
+      return vsGet(`${BOARD_AUX}.${slot}`);
+    };
+
+    const { entries } = Object;
+    const uiBridge = Far('UIBridge', {
+      /** @param {import('@endo/marshal').CapData<string>} offerEncoding */
+      proposeOffer: async offerEncoding => {
+        /** @type {import('../src/offers.js').OfferSpec} */
+        const offer = ctx.fromBoard.fromCapData(offerEncoding);
+        const { give, want } = offer.proposal;
+        for await (const [kw, amt] of entries({ ...give, ...want })) {
+          // @ts-expect-error
+          const { displayInfo } = await auxInfo(amt.brand);
+          const kind = displayInfo?.assetKind === 'nat' ? 'fungible' : 'NFT';
+          t.log('wallet:', kind, 'display for', kw, amt.brand);
+        }
+        t.log(addr, 'walletUI: approve offer:', offer.id);
+        await signAndBroadcast(harden({ method: 'executeOffer', offer }));
+      },
+    });
+    bridgeKit.resolve(uiBridge);
```

Comment:
why have `bridgeKit` instead of `uiBridge`?

Reply:
I think I don't understand the question.

`bridgeKit` is a promise (etc.) for when `uiBridge` is available.

Reply:
Understood. I'm questioning why they have different names. IIRC in the bootstrap we'd have the promise in the space have the name of the thing. The promise is for a `uiBridge` so I'd expect it to be named that. NBD